gems/aws-sdk-core/lib/seahorse/client/plugins/request_callback.rb

```diff
@@ -80,18 +80,16 @@ def call(context)
                 context.http_request.body,
                 callback
               )
-              add_event_listeners(context)
             end
-            @handler.call(context)
+            resp = @handler.call(context)
+            unwrap_callback_body(context)
```

Comment:
Does this need to happen all the time, or only if context[:on_chunk_sent] is sent?

Reply:
Only when on_chunk_sent is set, but unwrap already does a check, so not sure I want to nest this in another if?

Reply:
Edit: did a bit of cleanup, not sure I have a preference for it or the other.